BigFileMergeSort/src/ua/goit/alg/BigFileMergeSort.java

```diff
@@ -0,0 +1,182 @@
+		if (partList.size() % 2 == 1) {
+			File merged = mergeFiles(partList.get(0), partList.get(1));
+			partList.add(merged);
+			partList.remove(1);
+			partList.remove(0);
+		}
+		for (int i = 0; i < partList.size()-1; i+=2) {
+			resultList.add(mergeFiles(partList.get(i), partList.get(i+1)));
+		}
+		resultList = mergeParts(resultList);
+		return resultList;
+	}
+
+	private static ArrayList<File> splitSort(File file, int blocksize) throws IOException {
+		int counterOfElements = 0;
+		int tmpint = 0;
+		Scanner sc = new Scanner(file);
+		ArrayList<File> filelist = new ArrayList<File>();
+		while (sc.hasNextInt()) { 
+			int[] array = new int[blocksize]; 
+			int i = 0;
+			while ((i < blocksize) && sc.hasNextInt()){
+				tmpint = sc.nextInt();
+				array[i] = tmpint;
+				counterOfElements++;
+				i++;
+			}
+			Arrays.sort(array);
+			filelist.add(writeToTempFile(array));
+		} 
+		sc.close();
+		removeExtraZeros(blocksize, counterOfElements, filelist);
+		return filelist; 
+	}
+
+	private static void removeExtraZeros(int blocksize, int counterOfElements,
+			ArrayList<File> filelist) throws FileNotFoundException, IOException {
+
+		if (counterOfElements % blocksize != 0) {
+			int modulo_start = (blocksize - counterOfElements % blocksize) * 4; 
+			int modulo_end = (blocksize) * 4 - 4;
+			int[] tmparray = new int[counterOfElements % blocksize];
+			int fileNumber = filelist.size() - 1;
+			File fileWhithOverZeros = filelist.get(fileNumber);
+			RandomAccessFile run1 = new RandomAccessFile(fileWhithOverZeros, "rw");
+			int tmpint;
+			int j = 0;
+			try {
+				while (modulo_start <= modulo_end)  {
+					tmpint = readIntFromFile(run1, modulo_end);
+					tmparray[j] = tmpint;
+					j++;
+					modulo_end -= 4;
+				}
+			} finally {
+				run1.close();
+			}
+			filelist.remove(fileNumber);
+			filelist.add(writeToTempFile(tmparray));
+		}
+	}  
+
+	private static File writeToTempFile(int[] array) throws IOException {
+		DataOutputStream fdo = null;
+		File newtmpfile = File.createTempFile("text", ".txt", new File(tempDir));
+		newtmpfile.deleteOnExit(); 
+		try {
+			fdo = new DataOutputStream(new FileOutputStream(newtmpfile));
+			for(int i = 0; i < array.length; i++) {
+				fdo.writeInt(array[i]);
+			}
+		} finally {
+			fdo.close();
+		}
+		return newtmpfile;
+	}
+
+	private static int readIntFromFile(RandomAccessFile run, long pointer) throws IOException {
+		run.seek(pointer);
+		int i = run.readInt();
+		return i;
+	}
+
+	private static void writeIntToFile(RandomAccessFile run, int i, long pointer) throws IOException {
+		run.seek(pointer);
+		run.writeInt(i);
+	}
+
+	private static File mergeFiles(File firstFile, File secondFile) throws IOException {
+		int firstInt; 
+		int secondInt; 
+		long firstPos = 0;
+		long secondPos = 0; 
+		long writePos = 0; 
+
+		File file = File.createTempFile("TMPFile", ".txt", new File(tempDir));
+		file.deleteOnExit(); 
+		RandomAccessFile run1 = new RandomAccessFile(firstFile, "rw");
```

Comment:
please name vars somewhat explanatory, like source1, source2, result etc. also if you only reading file, please open it just for reading ("r")